src/m365/pp/commands/managementapp/managementapp-add.ts

```diff
@@ -0,0 +1,132 @@
+import { Application } from '@microsoft/microsoft-graph-types';
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+export interface Options extends GlobalOptions {
+  appId?: string;
+  objectId?: string;
+  name?: string;
+}
+
+class PpManagementAppAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.MANAGEMENTAPP_ADD;
+  }
+
+  public get description(): string {
+    return 'Register management application for Power Platform';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.appId = typeof args.options.appId !== 'undefined';
+    telemetryProps.objectId = typeof args.options.objectId !== 'undefined';
+    telemetryProps.name = typeof args.options.name !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getAppObjectId(args)
+      .then((appId: string): Promise<any> => {
+        const requestOptions: any = {
+          // This should be refactored once we implement a PowerPlatform base class as api.bap will differ between envs.
+          url: `https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/adminApplications/${appId}?api-version=2020-06-01`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.put(requestOptions);
+      })
+      .then((res: any): void => {
+        logger.log(res);
+        cb();
+      }, (err: any) => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getAppObjectId(args: CommandArgs): Promise<string> {
+    if (args.options.appId) {
```

Comment:
This is incorrect. Since we're looking up `objectId`, we shouldn't early exit when the user specified `appId`